From 9b60e49a6f0051537e0fad0fea908698f81b9434 Mon Sep 17 00:00:00 2001
From: Yonghua Zheng <yh.zheng@yuneec.com>
Date: Wed, 3 Mar 2021 23:16:59 +0800
Subject: [PATCH] Add wait for state change of fakesrc

This is to ensure dummy packet is sent before responding with PLAY
back to RTSP server to make sure UDP packets to traverse firewalls
and NAT
---
 gst/rtsp/gstrtspsrc.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gst/rtsp/gstrtspsrc.c b/gst/rtsp/gstrtspsrc.c
index cc97bf7..bba2717 100644
--- a/gst/rtsp/gstrtspsrc.c
+++ b/gst/rtsp/gstrtspsrc.c
@@ -4693,6 +4693,9 @@ gst_rtspsrc_send_dummy_packets (GstRTSPSrc * src)
     gst_element_set_state (stream->rtpsrc, GST_STATE_NULL);
     gst_element_set_state (stream->udpsink[0], GST_STATE_PLAYING);
     gst_element_set_state (stream->rtpsrc, GST_STATE_PLAYING);
+
+    /* some intensional sleep here to make sure dummy packet sent */
+    g_usleep(100*1000);
   }
   return TRUE;
 }
-- 
2.25.1

